compiler/crates/relay-bin/src/main.rs

```diff
@@ -197,6 +237,51 @@ async fn main() {
     }
 }
 
+async fn handle_lsp_command(command: LspCommand) {
+    configure_logger(command.output, TerminalMode::Stderr);
+
+    let config = get_config(command.config, command.cli_config);
+
+    let perf_logger = Arc::new(ConsoleLogger);
+    let extra_data_provider = Box::new(DummyExtraDataProvider::new());
+    let schema_documentation_loader: Option<Box<dyn SchemaDocumentationLoader<SDLSchema>>> = None;
+    let js_language_server = None;
+
+    match start_language_server(
+        config,
+        perf_logger,
+        extra_data_provider,
+        schema_documentation_loader,
+        js_language_server,
+    )
+    .await
+    {
+        Ok(_) => {
+            info!("Relay LSP exited successfully.");
+        }
+        Err(err) => {
+            error!("Relay LSP unexpectedly terminated: {:#?}", err);
+            std::process::exit(1);
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() {
```

Comment:
Nit. Can we move this to the top, and have the supporting functions below (I know this is code preference, but I think it makes it easier to quickly understand what the module is trying to do.